synapse/appservice/__init__.py

```diff
@@ -125,6 +127,17 @@ def _check_namespaces(self, namespaces):
                     raise ValueError(
                         "Expected bool for 'exclusive' in ns '%s'" % ns
                     )
+                if regex_obj.get("group_id"):
+                    if not isinstance(regex_obj.get("group_id"), str):
+                        raise ValueError(
+                            "Expected string for 'group_id' in ns '%s'" % ns
+                        )
+                    if not ApplicationService.GROUP_ID_REGEX.match(
+                            regex_obj.get("group_id")):
+                        raise ValueError(
+                            "Expected valid group ID for 'group_id' in ns '%s'" % ns
+                        )
```

Comment:
Just try and parse it as a `synapse.types.GroupID` instead of trying to come up with your own validation regex.

We should also check the server_name, which will probably require us to either pass in the `hs` object or `server_name` to the constructor.